### ISSUE TITLE
na_ontap_qtree : KeyError: No element by given name mode.

### ISSUE THREAD

**Issue by hellsbunny** [ansible-collections/netapp#75]
##### SUMMARY
The playbook can run only once. One complete run, a second run ends with module error.

##### ISSUE TYPE
after first run, the module na_ontap_qtree then looks for a key named "mode" to fill variable unix_permissions, but this keyname does not exists.

Workaround:  disabled line 186 in na_ontap_qtree.py as key mode does not exists
```
 return_q = {'export_policy': result['attributes-list']['qtree-info']['export-policy'],
#            'unix_permissions': result['attributes-list']['qtree-info']['mode'],
             'oplocks': result['attributes-list']['qtree-info']['oplocks'],
             'security_style': result['attributes-list']['qtree-info']['security-style']}
```

##### COMPONENT NAME
Line 186 /usr/lib/python2.7/dist-packages/ansible/modules/storage/netapp/na_ontap_qtree.py

##### ANSIBLE VERSION
```
ansible 2.9.12
```

##### CONFIGURATION
```
netapp-lib (2018.11.13) # /usr/local/lib/python2.7/dist-packages/netapp_lib

# ansible-config dump --only-changed
COLLECTIONS_PATHS(/etc/ansible/ansible.cfg) = [u'/etc/ansible/collections', u'/etc/ansible/collections', u'/etc/ansible/collections/ansible_collections']
DEFAULT_GATHERING(/etc/ansible/ansible.cfg) = explicit
DEFAULT_REMOTE_USER(/etc/ansible/ansible.cfg) = ansible
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /etc/ansible/.ansible_vault
DEPRECATION_WARNINGS(/etc/ansible/ansible.cfg) = False
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT
```
ONTAP: NetApp Release 9.7
OS: Linux ansible01 4.15.0-54-generic #58-Ubuntu SMP
```
##### STEPS TO REPRODUCE
Call this task twice, assuming that u have a dummy volume/export_policy already
```
    - name: Create Qtree
      na_ontap_qtree:
        hostname: "{{ hostname }}"
        username: "{{ username }}"
        password: "{{ password }}"
        state: "present"
        name: "tq1"
        flexvol_name: "testqtree1"
        # Workaround: disable na_ontap_qtree Line 186 (unix_premissions= KeyError mode)
        export_policy: "nfs-testqtree1"
        security_style: "ntfs"
        oplocks: enabled
        unix_permissions: ""
        vserver: "svm_test"
```

##### EXPECTED RESULTS
On first run alls seems ok.
On second run I expected, to run without error, as nohting changed,
but getting module error below.
KeyError: 'No element by given name mode.'\n",

##### ACTUAL RESULTS

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'No element by given name mode.'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1608029769.31-131028-78894809548122/AnsiballZ_na_ontap_qtree.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1608029769.31-131028-78894809548122/AnsiballZ_na_ontap_qtree.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1608029769.31-131028-78894809548122/AnsiballZ_na_ontap_qtree.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.storage.netapp.na_ontap_qtree', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib/python2.7/runpy.py\", line 188, in run_module\n    fname, loader, pkg_name)\n  File \"/usr/lib/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/usr/lib/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/tmp/ansible_na_ontap_qtree_payload_2mH9bD/ansible_na_ontap_qtree_payload.zip/ansible/modules/storage/netapp/na_ontap_qtree.py\", line 303, in <module>\n  File \"/tmp/ansible_na_ontap_qtree_payload_2mH9bD/ansible_na_ontap_qtree_payload.zip/ansible/modules/storage/netapp/na_ontap_qtree.py\", line 299, in main\n  File \"/tmp/ansible_na_ontap_qtree_payload_2mH9bD/ansible_na_ontap_qtree_payload.zip/ansible/modules/storage/netapp/na_ontap_qtree.py\", line 273, in apply\n  File \"/tmp/ansible_na_ontap_qtree_payload_2mH9bD/ansible_na_ontap_qtree_payload.zip/ansible/modules/storage/netapp/na_ontap_qtree.py\", line 186, in get_qtree\n  File \"/usr/local/lib/python2.7/dist-packages/netapp_lib/api/zapi/zapi.py\", line 489, in __getitem__\n    raise KeyError('No element by given name %s.' % key)\nKeyError: 'No element by given name mode.'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Comment by carchi8py** [comment]
@hellsbunny could you try using the Collection version of the modules. the 2.9 version of the modules are very old at this point. There steps on how to install and use the collection here. 

https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/ontap/README.md

**Comment by lonico** [comment]
Yes, this was fixed in 19.10.0
```
na_ontap_qtree: unix permission is not available when security style is ntfs
```

**Comment by hellsbunny** [comment]
ok thanks  works as described,... 

---
installed galaxy collection version and updated all roles:

<ROLENAME>/meta/main.yaml
```
collections:
  - netapp.ontap
```
